tfhe/src/shortint/ciphertext/mod.rs

```diff
@@ -1,897 +1,10 @@
-
-        assert_eq!(max_noise_level.0, 5);
-    }
-
-    #[test]
-    fn degree_after_bitxor_ci_run_filter() {
-        let data = [
-            (Degree(3), Degree(3), Degree(3)),
-            (Degree(3), Degree(1), Degree(3)),
-            (Degree(1), Degree(3), Degree(3)),
-            (Degree(3), Degree(2), Degree(3)),
-            (Degree(2), Degree(3), Degree(3)),
-            (Degree(2), Degree(2), Degree(3)),
-            (Degree(2), Degree(1), Degree(3)),
-            (Degree(1), Degree(2), Degree(3)),
-            (Degree(1), Degree(1), Degree(1)),
-            (Degree(0), Degree(1), Degree(1)),
-            (Degree(0), Degree(1), Degree(1)),
-        ];
-
-        for (lhs, rhs, expected) in data {
-            let result = lhs.after_bitxor(rhs);
-            assert_eq!(
-                result, expected,
-                "For a bitxor between variables of degree {lhs:?} and {rhs:?},\
-             expected resulting degree: {expected:?}, got {result:?}"
-            );
-        }
-    }
-    #[test]
-    fn degree_after_bitor_ci_run_filter() {
-        let data = [
-            (Degree(3), Degree(3), Degree(3)),
-            (Degree(3), Degree(1), Degree(3)),
-            (Degree(1), Degree(3), Degree(3)),
-            (Degree(3), Degree(2), Degree(3)),
-            (Degree(2), Degree(3), Degree(3)),
-            (Degree(2), Degree(2), Degree(3)),
-            (Degree(2), Degree(1), Degree(3)),
-            (Degree(1), Degree(2), Degree(3)),
-            (Degree(1), Degree(1), Degree(1)),
-            (Degree(0), Degree(1), Degree(1)),
-            (Degree(0), Degree(1), Degree(1)),
-        ];
-
-        for (lhs, rhs, expected) in data {
-            let result = lhs.after_bitor(rhs);
-            assert_eq!(
-                result, expected,
-                "For a bitor between variables of degree {lhs:?} and {rhs:?},\
-             expected resulting degree: {expected:?}, got {result:?}"
-            );
-        }
-    }
-
-    #[test]
-    fn degree_after_bitand_ci_run_filter() {
-        let data = [
-            (Degree(3), Degree(3), Degree(3)),
-            (Degree(3), Degree(1), Degree(1)),
-            (Degree(1), Degree(3), Degree(1)),
-            (Degree(3), Degree(2), Degree(2)),
-            (Degree(2), Degree(3), Degree(2)),
-            (Degree(2), Degree(2), Degree(2)),
-            (Degree(2), Degree(1), Degree(1)),
-            (Degree(1), Degree(2), Degree(1)),
-            (Degree(1), Degree(1), Degree(1)),
-            (Degree(0), Degree(1), Degree(0)),
-            (Degree(0), Degree(1), Degree(0)),
-        ];
-
-        for (lhs, rhs, expected) in data {
-            let result = lhs.after_bitand(rhs);
-            assert_eq!(
-                result, expected,
-                "For a bitand between variables of degree {lhs:?} and {rhs:?},\
-             expected resulting degree: {expected:?}, got {result:?}"
-            );
-        }
-    }
-}
+#[allow(clippy::module_inception)]
```

Comment:
unsure for the naming below

also, let's keep the modules public for now, more often than not having private stuff ends being more problems than it's worth at the moment

Reply:
By reexporting everything publicly, the interface of this module remains unchanged I believe
I don't see what problem this could cause

Reply:
we'll see but from experience on TFHE-rs as soon as something is not public, then it gets annoying/someone needs access for fairly good reason

Reply:
I believe everything is accessible through the `pub use`, the only difference being that it appears one module higher (as before this refactor)

Reply:
as I said, we will see :)